include/utility/json_utility.hpp

```diff
@@ -926,5 +926,68 @@ inline bool isFruPowerOffOnly(const nlohmann::json& i_sysCfgJsonObj,
             (i_sysCfgJsonObj["frus"][i_vpdFruPath].at(0)["powerOffOnly"]));
 }
 
+/**
+ * @brief API which tells if the FRU 's VPD is recollectable at runtime
+ *
+ * @param[in] i_sysCfgJsonObj - System config JSON object.
+ * @param[in] i_dbusObjPath - D-bus object path.
+ *
+ * @return true if FRU 's VPD is recollectable at runtime. false otherwise.
+ */
+inline bool isVPDRecollectableAtRuntime(const nlohmann::json& i_sysCfgJsonObj,
+                                        const std::string& i_dbusObjPath)
+{
+    try
+    {
+        if (i_dbusObjPath.empty())
+        {
+            throw std::runtime_error("Given D-bus object path is empty.");
+        }
+
+        if (i_sysCfgJsonObj.empty())
+        {
+            throw std::runtime_error(
+                "System config JSON object is not present. Failed checking if the FRU VPD is recollectable at runtime or not for " +
+                std::string(i_dbusObjPath));
+        }
+
+        if (!i_sysCfgJsonObj.contains("frus"))
```

Comment:
Should we check for both "frus" and emptiness w.r.t. JSON? Can we optimize this by just checking the "frus" part? Any downside there?